R/visualize_embeddings.R

```diff
@@ -143,77 +144,150 @@ keep_tokens <- function(embedding_df, tokens = "[CLS]") {
-                                -dplyr::matches("V[0-9]+"))
-
-    vec_mat <- as.matrix(
-        dplyr::select(embedding_df,
-                      dplyr::matches("V[0-9]+"))
-    )
-    pcs <- stats::prcomp(vec_mat,
-                         retx = TRUE, center = TRUE, scale. = TRUE, rank. = 2L)
-    pc_tbl <- dplyr::bind_cols(tok_labels, tibble::as_tibble(pcs$x))
-
-    class <- rep("a", num_rows)
-    if (!is.null(color_field)) {
-        if (color_field %in% names(embedding_df)) {
-            class <- dplyr::pull(
-                dplyr::select(embedding_df, dplyr::one_of(color_field))
-            )
-        } else {
-            warning("Column ", color_field, " not found in input table." )
-        }
-    }
-    class <- as.factor(class)
-
-    unique_classes <- unique(class)
-    num_class <- length(unique_classes)
-    getPalette <- grDevices::colorRampPalette(
-        RColorBrewer::brewer.pal(8, "Dark2")
-    )
-    pal <- getPalette(num_class)
-    class_colors <- pal
-    names(class_colors) <- unique_classes
-
-    if (!is.null(hide)) {
-        if (length(hide) == nrow(pc_tbl)) {
-            pc_tbl <- pc_tbl[!hide, ]
-            class <- class[!hide]
-        } else {
-            warning("Length of hide parameter doesn't match size of ",
-                    "input table, and will be ignored.")
-        }
-    }
+  if (!is.null(hide)) {
+    warning("`hide` has been deprecated. Use `project_vectors` to specify ",
+            "which vectors to use for the PCA calculation.")
+  }
 
-    ggp <- ggplot2::ggplot(pc_tbl, ggplot2::aes(x = PC1, y = PC2,
-                              label = token,
-                              col = class)) +
-        ggplot2::scale_color_manual(values = class_colors,
-                           name = color_field) +
-        ggplot2::geom_text(vjust = 0, nudge_y = 0.5) +
-        ggplot2::geom_point()
-    if (num_class <= 1) {
-        ggp <- ggp + ggplot2::theme(legend.position = "none")
-    }
-    return(ggp)
+  pca_df <- do_pca(embedding_df = embedding_df,
+                   project_vectors = project_vectors,
+                   color_field = color_field,
+                   disambiguate_tokens = disambiguate_tokens)
+  return(
+    plot_pca(pca_df = pca_df,
+             color_field = color_field)
+  )
 }
 
+# do_pca -------------------------------------------------------------
 
+#' Do PCA on Embeddings
+#'
+#' Calculate the 2D PCA for a collection of embedding vectors, in preparation
+#' for plotting.
+#'
+#' @inheritParams display_pca
+#'
+#' @return A tbl_df of the embedding vectors projected onto two principal
+#'   axes.
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' # assuming something like the following has been run:
+#' # feats <- RBERT::extract_features(...) # See RBERT documentation
+#' # Then:
+#' pca_df <- feats$output %>%
+#'     filter_layer_embeddings(layer_indices = 12L) %>%
+#'     keep_tokens("[CLS]") %>%
+#'     do_pca()
+#' }
+do_pca <- function(embedding_df,
```

Comment:
I recommend `calculate_pca_vectors` or `calculate_pca`. It's longer, but "do" is so generic.

Reply:
Yeah, in my head it became "do principle component analysis". Which reads better to me than "calculate ...". But I'm not at all happy with the names as they are. So still thinking about this.

Reply:
Realized we didn't resolve this yet. I have a mild preference for "do" in this case. Do you have a strong opinion about it?